frontend/src/components/globalSearch/GlobalSearchContent.tsx

```diff
@@ -175,7 +178,68 @@ export function GlobalSearchContent({
 
   // Resource search items
   const resources = useSearchResources();
+  const resourcesByKind = useMemo(
+    () => new Map(resources.map(resource => [resource.kind, resource])),
+    [resources]
+  );
   const loading = resources.filter(it => it.isLoading).map(it => it.kind);
+  const namespaceItems = (resourcesByKind.get(Namespace.kind)?.items ?? []) as Namespace[];
+  const namespaceOptions = useMemo(() => {
+    const knownNamespaces = new Set<string>();
```

Comment:
This is more of a styling comment, so feel free to ignore, but the same Set can be initialized more concisely 

```tsx
const knownNamespaces = new Set<string>([
  // Loaded namespaces
  ...namespaceItems.map(n => n.metadata.name),
  // Allowed namespaces from settings
  ...selectedClusters.flatMap(c => loadClusterSettings(c)?.allowedNamespaces)
].filter(Boolean));
```